repub.go

```diff
@@ -58,46 +58,57 @@ func (rp *Republisher) WaitPub(ctx context.Context) error {
 	}
 }
 
-func (rp *Republisher) Close() error {
-	// TODO(steb): Wait for `Run` to stop
-	err := rp.WaitPub(rp.ctx)
-	rp.cancel()
-	return err
+// Close tells the republisher to stop and waits for it to stop.  If it is
+// necessary to wait for any current values to be published, then call WaitPub
+// prior to calling Close.
```

Comment:
I get the logic but I'm not sure I understand this sentence. Are we suggesting the user to call `WaitPub` or is this a description of what we are doing internally?

Reply:
It is saying that the user should not expect `WaitPub()` to work after calling `Close()`, so if they need to use `WaitPub()` they need to finish doing that before calling `Close()`. I think that goes without saying, so I removed the comment.